app/helpers/deploys_helper.rb

```diff
@@ -1,8 +1,26 @@
 require 'coderay'
 
 module DeploysHelper
-  def deploy_active?
-    @deploy.waiting_for_buddy? || deploy_running? || deploy_queued?
+  def deploy_output
+    output_hidden = false
+    output = ActiveSupport::SafeBuffer.new
+
+    if JobExecution.enabled
+      output << Samson::Hooks.render_views(:deploy_view, self, deploy: @deploy, project: @project)
+
+      if deploy_queued?
+        output_hidden = true
+        output << render('queued')
+      elsif @deploy.waiting_for_buddy?
+        output_hidden = true
+        output << render('buddy_check', deploy: @deploy)
+      end
+    elsif @deploy.active?
```

Comment:
Job execution is disabled while the server is restarting but there are still jobs pending. This is actually the fix, to properly show a queued message when that happens.